File: src/algorithms/split-words.js
/**
 * Given an array of strings words and a character separator, splits each string
 * in words by separator and returns an array of strings containing the new
 * strings formed after the splits, excluding empty strings.
 * 
 * @param {string[]} words - Array of lowercase English letters and characters
 * @param {string} separator - A character from ".,|$#@" (excluding the quotes)
 * @returns {string[]} - Returns an array containing the new strings formed
 *     after the splits
 */
function splitWordsBySeparator (words, separator) {
  let res = [];
  for (let i = 0; i < words.length; i++) {
    let splitWords = words[i].split(separator);
    for (let j = 0; j < splitWords.length; j++) {
      res.push(splitWords[j]);
    }
  }
  res = res.filter(e => e);
  return res;
}

File: src/algorithms/split-words.py
def split_words_by_separator(words, separator):
    """Given an array of strings words and a character separator, split each 
    string in words by separator. Return an array of strings containing the new 
    strings formed after the splits, excluding empty strings.

    The separator is used to determine where the split should occur, but it is 
    not included as part of the resulting strings. A split may result in more 
    than two strings. The resulting strings must maintain the same order as 
    they were initially given.

    Args:
        words (list[str]): A list of strings.
        separator (str): A character from the string ".,|$#@".

    Returns:
        list[str]: Returns a list of strings.
    """
    res = []
    for i in range(len(words)):
        curr = words[i].split(separator)
        for word in curr:
            if len(word) > 0:
                res.append(word)
    return res
